src/rgw/rgw_civetweb_frontend.cc

```diff
@@ -64,6 +64,9 @@ int RGWCivetWebFrontend::run()
     conf_map["run_as_user"] = std::move(uid_string);
   }
 
+  /* Set authentication domains, this is needed for handling absolute uris in
+     requests */
+  conf_map["authentication_domain"] = g_conf->rgw_dns_name;
```

Comment:
@yehudasa this is the current behaviour of civetweb (with the proposed cherry-pick) https://github.com/civetweb/civetweb/issues/197#issuecomment-255186417